src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -836,6 +836,52 @@ void C2_MacroAssembler::fast_unlock_lightweight(Register obj, Register reg_rax,
   // C2 uses the value of ZF to determine the continuation.
 }
 
+#ifdef ASSERT
+void C2_MacroAssembler::checked_cast_int(const TypeInt* type, Register dst) {
```

Comment:
Naming is a bit confusing here. It is a register which holds the value being range checked, not a register where new value is put.

Reply:
I have renamed it to `val`